django/forms/widgets.py

```diff
@@ -1161,6 +1161,8 @@ def value_from_datadict(self, data, files, name):
                 # Return pseudo-ISO dates with zeros for any unselected values,
                 # e.g. '2017-0-23'.
                 return "%s-%s-%s" % (y or 0, m or 0, d or 0)
+            except OverflowError:
+                return "0-0-0"
```

Comment:
I think we should return `None` instead.

Reply:
I don't think so, if you return `None` here, then the form error message will be
'This field is required.' instead of 'Enter a valid date.', while the latter is actually describing what is wrong (the field is not empty, the date is invalid). I chose '0-0-0' because it's consistent with the other exceptional case, where '0-0-0' is returned when `y`, `m` and `d` are all invalid.